### PR TITLE
feat: profile page github activity graph

### PR DESCRIPTION
**Pull request by thititongumpun**
This pull request introduces a feature to the /profile page
– The addition activity calendar toggle button . This toggle button empowers users to display github activity (contributions) calendar in their card profile. 

image

image

### CODE REVIEW

File: src/pages/profile/index.tsx [hunk 1]
```diff
@@ -6,16 +6,42 @@ import { useGitHubQuery } from "@/hooks";
...
+
+interface Activity {
+  date: string;
+  count: number;
+  level: 0 | 1 | 2 | 3 | 4;
+}
 
 export default function Profile() {
   const { data } = useGitHubQuery<UserProfile>(userProfileQuery);
+  const [checked, setChecked] = useState<boolean>(false);
```
Comment (Balastrong) [inline src/pages/profile/index.tsx:20]: Can we use a more self explanatory variable here? `checked` is too generic

File: src/pages/profile/index.tsx [hunk 2]
```diff
@@ -6,16 +6,42 @@ import { useGitHubQuery } from "@/hooks";
...
+      const date = new Date(activity.date);
+      const monthOfDay = date.getMonth();
+
+      return (
+        date.getFullYear() === currentYear &&
+        monthOfDay > currentMonth - shownMonths &&
+        monthOfDay <= currentMonth
+      );
+    });
+  };
```
Comment (Balastrong) [inline src/pages/profile/index.tsx:39]: This feels like it should be simplified somehow.
A hint, can we define a `startDate` that is 6 months from now, and then just make sure that each `activity.date` is after that?

File: src/pages/profile/index.tsx [hunk 3]
```diff
@@ -81,6 +118,20 @@ export default function Profile() {
...
+                <GitHubCalendar
+                  username={data.user.login}
+                  transformData={selectLastHalfYear}
+                  blockSize={10}
+                  fontSize={10}
+                  loading={!data}
+                  colorScheme="light"
+                  hideColorLegend
+                  labels={{
+                    totalCount: "{{count}} contributions in the last half year",
```
Comment (Balastrong) [inline src/pages/profile/index.tsx:131]:
```suggestion
                    totalCount: "{{count}} contributions in the last 6 months",
```

I think with "6 months" is more readable

**Review by Balastrong — CHANGES_REQUESTED** [review]
Thanks for the PR, I like this widget!

I added a few comments, can you please review them?

File: src/pages/profile/index.tsx (hunk 3 above)
Comment (thititongumpun) [reply src/pages/profile/index.tsx:131]: lgtm

File: src/pages/profile/index.tsx (hunk 1 above)
Comment (thititongumpun) [reply src/pages/profile/index.tsx:20]: Agreed.

File: src/pages/profile/index.tsx (hunk 2 above)
Comment (thititongumpun) [reply src/pages/profile/index.tsx]: I defined it's.

File: src/pages/profile/index.tsx [hunk 4]
```diff
@@ -6,16 +6,46 @@ import { useGitHubQuery } from "@/hooks";
...
+
+    return contributions.filter((activity: Activity) => {
+      const date = new Date(activity.date);
+      const monthOfDay = date.getMonth();
+
+      return (
+        date.getFullYear() === currentYear &&
+        monthOfDay > currentMonth - shownMonths &&
+        monthOfDay <= currentMonth &&
+        startDate.getTime() <= new Date(activity.date).getTime()
```
Comment (Balastrong) [inline src/pages/profile/index.tsx:40]: Now that we have `startDate.getTime() <= new Date(activity.date).getTime()` do you think we still need the other checks or are they redundant and can be removed?

Actually `date.getFullYear() === currentYear` looks wrong, if we're in february and we check for same year, only two months will be displayed as the other 4 have a different year.

Comment (thititongumpun) [reply src/pages/profile/index.tsx:40]: > Now that we have `startDate.getTime() <= new Date(activity.date).getTime()` do you think we still need the other checks or are they redundant and can be removed?
> 
> Actually `date.getFullYear() === currentYear` looks wrong, if we're in february and we check for same year, only two months will be displayed as the other 4 have a different year.

```suggestion
        monthOfDay > currentMonth - shownMonths &&
        monthOfDay <= currentMonth &&
        startDate.getTime() <= new Date(activity.date).getTime()
```

You're absolutely right  `date.getFullYear() === currentYear` looks wrong 
`monthOfDay > currentMonth - shownMonths &&  monthOfDay <= currentMonth` still need for get between in range Correct me if I'm wrong.

Comment (Balastrong) [reply src/pages/profile/index.tsx]: If we want the data to be "the last full 6 months" then we need it, but if we want "the last 6 months from today (6th october -> 6th April) then the `.getTime()` check is enough.

Let's think of what design choice makes more sense, if full months or months from current date 🤔

Comment (thititongumpun) [reply src/pages/profile/index.tsx]: In my opinion maybe display months from current date, Simplify subtracts from 6 months.

image

Comment (Balastrong) [reply src/pages/profile/index.tsx]: I agree! Then the getTime() check should be enough :)

**Review by thititongumpun — COMMENTED** [review]
Thanks.

**Review by Balastrong — APPROVED** [review]
LGTM, thank you for the PR and for considering the suggestions! :D

**Comment by Balastrong** [comment]
@all-contributors please add @thititongumpun for code

**Pull request merged**